kloppy/infra/serializers/tracking/statsperform.py

```diff
@@ -0,0 +1,271 @@
+            )
+            ball_coordinates = Point3D(ball_x, ball_y, ball_z)
+        else:
+            ball_coordinates = None
+
+        players_data = {}
+        player_info = components[1]
+        for player_data in player_info.split(";")[:-1]:
+            (
+                team_side_id,
+                player_id,
+                jersey_no,
+                x,
+                y,
+            ) = player_data.split(",")
+
+            # Goalkeepers have id 3 and 4
+            if int(team_side_id) > 2:
+                team_side_id = int(team_side_id) - 3
+            team = teams[int(team_side_id)]
+            player = team.get_player_by_id(player_id)
+
+            if not player:
+                player = Player(
+                    player_id=player_id,
+                    team=team,
+                    jersey_no=int(jersey_no),
+                )
+                team.players.append(player)
+
+            players_data[player] = PlayerData(
+                coordinates=Point(float(x), float(y))
+            )
+
+        return Frame(
+            frame_id=frame_id,
+            timestamp=frame_timestamp,
+            ball_coordinates=ball_coordinates,
+            ball_state=ball_state,
+            ball_owning_team=ball_owning_team,
+            players_data=players_data,
+            period=period,
+            other_data={},
+        )
+
+    @staticmethod
+    def __validate_inputs(inputs: Dict[str, Readable]):
+        if "xml_metadata" not in inputs:
+            raise ValueError("Please specify a value for 'xml_metadata'")
+        if "raw_data" not in inputs:
+            raise ValueError("Please specify a value for 'raw_data'")
+
+    def deserialize(self, inputs: StatsperformInputs) -> TrackingDataset:
+        raw_data = inputs.raw_data.read()
+        metadata = inputs.meta_data.read()
+
+        with performance_logging("Loading XML metadata", logger=logger):
+            match = objectify.fromstring(metadata).matchInfo
+            teams = []
+            for team_info in match.contestants.iterchildren(tag="contestant"):
+                if team_info.attrib["position"] == "home":
+                    ground = Ground.HOME
+                else:
+                    ground = Ground.AWAY
+                team = Team(
+                    team_id=team_info.attrib["id"],
+                    name=team_info.attrib["name"],
+                    ground=ground,
+                )
+                teams.append(team)
+
+            # Get Frame Rate - Either 10FPS or 25FPS
+            frame_rate = self.__get_frame_rate(raw_data)
+            pitch_size_length = 100
+            pitch_size_width = 100
+
+            periods = self.__get_periods(raw_data)
+            # Get Player Info:
+            for line_up in objectify.fromstring(
+                metadata
+            ).liveData.iterchildren(tag="lineUp"):
+                count = 0
+                for team in teams:
```

Comment:
I would use `enumerate` instead of manually keeping track of the `count` variable.

However, I think an even better solution would be to turn the `teams` `list` into a `dict` where the key is the team ID and the value is the `Team` object.

```python
team_id = line_up.get("contestantId")
team = teams[team_id]

teams_list = list(teams.values())
metadata = Metadata(
    teams=teams_list,
    ...
)
```